data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt

```diff
@@ -0,0 +1,276 @@
+        amount: String, srcAddress: String, isAffiliate: Boolean
+    ): KyberSwapQuoteDeserialized {
+        val sourceAddress =
+            if (srcTokenContractAddress.isEmpty()) NULL_ADDRESS else srcTokenContractAddress
+        val destinationAddress =
+            if (dstTokenContractAddress.isEmpty()) NULL_ADDRESS else dstTokenContractAddress
+
+        val response = httpClient.get(aggregatorApiBaseUrl) {
+            url {
+                path(
+                    chain.raw.lowercase(),
+                    "api/v1/routes",
+                )
+                parameters.append(
+                    "tokenIn",
+                    sourceAddress
+                )
+                parameters.append(
+                    "tokenOut",
+                    destinationAddress
+                )
+                parameters.append(
+                    "amountIn",
+                    amount
+                )
+                parameters.append(
+                    "saveGas",
+                    "false"
+                )
+                parameters.append(
+                    "gasInclude",
+                    "true"
+                )
+                parameters.append(
+                    "slippageTolerance",
+                    "100"
+                )
+                parameters.append(
+                    "isAffiliate",
+                    isAffiliate.toString()
+                )
+                parameter(
+                    "sourceIdentifier",
+                    if (isAffiliate) CLIENT_ID else null
+                )
+                parameter(
+                    "referrerAddress",
+                    if (isAffiliate) REFERRER_ADDRESS else null
+                )
+
+                headers {
+                    accept(ContentType.Application.Json)
+                    append(
+                        "x-client-id",
+                        CLIENT_ID
+                    )
+                }
+            }
+        }
+
+        val responseString = response.body<String>()
+
+        if (!response.status.isSuccess()) {
+            return KyberSwapQuoteDeserialized.Error(
+                error = HttpStatusCode.fromValue(response.status.value).description
+            )
+        }
+        val errorResponse = runCatching {
+            json.decodeFromString<KyberSwapErrorResponse>(responseString)
+        }.getOrNull()
+
+        if (errorResponse != null && errorResponse.code != 0) {
+            throw KyberSwapError.ApiError(
+                errorResponse.code,
+                errorResponse.message,
+                errorResponse.details
+            )
+        }
+
+        val route = json.decodeFromString<KyberSwapRouteResponse>(responseString)
+        return KyberSwapQuoteDeserialized.Result(
+            buildTransactionWithFallback(
+                chain,
+                route,
+                srcAddress,
+                isAffiliate
+            )
+        )
```

Comment:
Do we care about all these error handling? or can we just do bodyOrThrow() and throw exception or map to exception with specific message?